### ISSUE TITLE
Config request: eval custom code on loading a new file

### ISSUE THREAD

**Issue by sindhubb** [BetterThanTomorrow/calva#2180]
Currently, `afterCLJReplJackInCode` executes first time the REPL is loaded and not whenever a new file is evaluated. 
`evalOnSave` setting enables evaluating whenever it is saved but skips the code in after `afterCLJReplJackInCode`. 

Can we have a new config which executes for every file for example: one which requires `required` symbols. 

I built a custom repl as a workaround with the following middleware. But I would prefer if Calva directly supports it. 

```clj
(defn eval-in-session [msg eval-fn form]
  (let [response (eval-fn (assoc msg
                                 :op "eval"
                                 :code (pr-str form)))]
    (when (:err response)
      (throw (ex-info "Error evaluating form in session" {:form form :response response})))))

(defn myproject-custom-middleware [handler]
  (fn [msg]
    (when (#{"init-debugger" "load-file"} (:op msg))
      (eval-in-session msg handler `(require '[myproject.debug.repl-req :refer :all])))
    (handler msg)))
```

**Comment by PEZ** [comment]
We can certainly consider. Meanwhile, it is pretty easy to solve this using [Joyride](https://github.com/BetterThanTomorrow/joyride) and [Calva's REPL API](https://calva.io/api/#replevaluatecode). Please let me know if you want to try this and need some pointers.

**Comment by sindhus-in** [comment]
> We can certainly consider. Meanwhile, it is pretty easy to solve this using [Joyride](https://github.com/BetterThanTomorrow/joyride) and [Calva's REPL API](https://calva.io/api/#replevaluatecode). Please let me know if you want to try this and need some pointers.

I would like to try it. Please let me know how.

**Comment by PEZ** [comment]
Hi again! Sorry for the radio silence. I decided to give your original suggestion a go and have a dev version of Calva where there is a setting added for running code when a file has been loaded. Here's a VSIX package with this. It's not quite finished yet, but please give it a spin and see if it solves your use case:

* VSIX: https://output.circle-artifacts.com/output/job/d2f7c20f-eaa2-4778-9f5d-3671390d145b/artifacts/0/tmp/artifacts/calva-2.0.356-2180-autoconnect-autoload-code-7ae65f3a.vsix

In VS Code's Extension pane, there's a small ”meatballs” menu at the top with the option to **Install from VSIX...** (Also, when you download that file, it might be saved with `.zip` extension, and then VS Code won't let you install it. If this happens, just change the extension to `.vsix`.)

There are no docs yet... But I think that if you search for the settings `calva.autoEvaluateCode` in the Settings view, you will probably figure it out. 😄 

I'll come back to you about the Joyride way of doing this. Need to go afk for a while here now.

**Comment by sindhubb** [comment]
> I'll come back to you about the Joyride way of doing this. Need to go afk for a while here now.
I wouldn't be needing this anymore. Thank you so much for introducing the  now that you merged #2181! 

This makes my life so much easier!
